src/abstractions/serialization/KiotaJsonSerializer.Deserialization.Untyped.cs

```diff
@@ -0,0 +1,88 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+
+#if NET5_0_OR_GREATER
+
+using System.Threading.Tasks;
+using System.Threading;
+using System.IO;
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics.CodeAnalysis;
+using System.Collections.Generic;
+using System.Linq;
+
+#pragma warning disable IL3050 
+
+namespace Microsoft.Kiota.Abstractions.Serialization;
+
+public static partial class KiotaJsonSerializer
+{
+    private static bool IsIParsable(this Type? type) => type?.IsAssignableTo(typeof(IParsable)) ?? false;
+    private abstract class KiotaJsonDeserializer
```

Comment:
can we give different names to this class and the one bellow to ease up code review?
It feels like this one should be called somethingFactory.
And the one at line 30 somethingAdapter.

Or something alone those lines.

Reply:
Hi baywet,

surely you could name them as you please, but to comment on your suggestions.
The factory is in fact an abstract base class having a factory method (well one line of it) and executes the actual Activator, which is then placed in line 30.

I created this much like in System.Security, where (f. e.) one calls the create method of the abstract AES, which returns an actual implementation of a derived class.

I tried to keep it as separate as possible and even the derived classes are privately nested classes of the abstract base.

If you want to call the class something different, you might call it a wrapper around the KitoJsonSerializer for this special case of having the target type as Type instead of handing it to a generic method. So I guess I will rename it.

I also got e broken build, which was the result of some wrong line endings and different whitespacing.
Is there something I could download for VS to have the style guide enforced before pushing it again to GitHub?